apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -509,6 +511,38 @@ object ConvertValueToDocumentReferencesForInQuery()
             }
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Query Where(Filter filter)
+        {
+            ValidateFilterForSentinels(filter);
+            var newFilter = (_filter == null) ? new Filter.CompositeFilter(new List<Filter>(), CompositeFilter.Types.Operator.And)
+            .WithAdditionalFilter(filter) : _filter.WithAdditionalFilter(filter);
+            return new Query(_root, _offset, _limit, _orderings, _projections, _startAt, _endAt, newFilter);
+        }
+
+        private void ValidateFilterForSentinels(Filter filter)
+        {
+            if (filter.GetType() == typeof(Filter.UnaryFilter))

```

Comment:
Calling `GetType()` is usually a design smell. I think it would make more sense to have an abstract `ValidaterForSentinels` method in `Filter`.